File: normal/diffLang/pathSum.go
package main

// Definition for a binary tree node.
type TreeNode struct {
    Val   int
    Left  *TreeNode
    Right *TreeNode
}

func hasPathSum(root *TreeNode, targetSum int) bool {
    if root == nil {
        return false // Base case: empty tree.
    }

    // Check if it's a leaf node and the value matches the remaining targetSum.
    if root.Left == nil && root.Right == nil && root.Val == targetSum {
        return true
    }

    // Recurse for left and right subtrees with the updated targetSum.
    targetSum -= root.Val
    return hasPathSum(root.Left, targetSum) || hasPathSum(root.Right, targetSum)
}

File: normal/diffLang/pathSum.ex
# Definition for a binary tree node.
defmodule TreeNode do
  defstruct val: 0, left: nil, right: nil
end

defmodule Solution do
  def has_path_sum(nil, _target_sum), do: false

  def has_path_sum(%TreeNode{val: val, left: nil, right: nil}, target_sum) do
    val == target_sum
  end

  def has_path_sum(%TreeNode{val: val, left: left, right: right}, target_sum) do
    new_target_sum = target_sum - val
    has_path_sum(left, new_target_sum) or has_path_sum(right, new_target_sum)
  end
end
